certbot/cli.py

```diff
@@ -1003,6 +1003,16 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
              "certificate already exists for the requested certificate name "
              "but does not match the requested domains, renew it now, "
              "regardless of whether it is near expiry.")
+    helpful.add(
+        "automation", "--reuse-key", dest="reuse_key",
+        action="store_true", default=flag_default("reuse_key"),
+        help="When renewing, use the same private key as the existing "
+             "certificate.")
+    helpful.add(
+        "automation", "--no-reuse-key", dest="reuse_key",
+        action="store_false", default=flag_default("reuse_key"),
```

Comment:
did we ever figure out why this works? as in where the magic code lives. just out of curiosity

Reply:
Nope, I'm not really sure!

Reply:
I'm not sure if this exactly answers your question, but we have a number of flags like `--foo` and `--no-foo`. For example, take a look at [\-\-redirect](https://github.com/certbot/certbot/blob/36dfd065037678bf38b7adb95ab16ed951ebbd18/certbot/cli.py#L1110).

`--redirect` sets `redirect` on the config object to `True` while `--no-redirect` sets it to `False`. Both `--redirect` and `--no-redirect` share the same default which isn't `True` or `False`, but [None](https://github.com/certbot/certbot/blob/36dfd065037678bf38b7adb95ab16ed951ebbd18/certbot/constants.py#L59). This allows us to differentiate between the user saying yes or no to a redirect and not answering the question.

Do we want to do anything special in the case where the user doesn't provide either `--reuse-key` or `--no-reuse-key`? If not, I think we can probably get rid of `--no-reuse-key` and keep the default `reuse_key` value to `False`. If we do want this for some reason, we should probably change the default value from `False` to `None`.

Reply:
After talking to @bmw, I more clearly understand the point that for options that don't relate to prompts, we don't need a separate `--no` version because simply leaving the option out when reissuing with `certonly` is already sufficient to set it to `False`.